Add ability to donate to Mastodon from the app

whattherestimefor opened the pull request:
This change only affects users logged in to mastodon.social or mastodon.online. A banner may be periodically displayed at the bottom of the homescreen encouraging donations and menu options are now available in Settings to make new donations or manage existing ones.

Completes IOS-252

![DonationsScreenshots 001](https://github.com/user-attachments/assets/dc1b731d-7fe4-4f31-9c39-fa870ca88f12)
![DonationsScreenshots 002](https://github.com/user-attachments/assets/386b1bc7-9268-42d1-b41e-63e7ec14a978)

kimar committed:
Implement Donations API, Data Model, API Service (IOS-252)

kimar committed:
Implement fetching of active donation campaign (IOS-252)

kimar committed:
Implement DonationBanner (IOS-252)

kimar committed:
Show Donation Sheet (IOS-252)

kimar committed:
Begin implementing Donation Modal (IOS-252)

whattherestimefor committed:
Add ability to donate to Mastodon from the app

This change only affects users logged in to mastodon.social or mastodon.online. A banner may be periodically displayed at the bottom of the homescreen encouraging donations and menu options are now available in Settings to make new donations or manage existing ones.

Applies swift-format to added files.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/mastodon/mastodon-ios?pullRequest=1353) <br/>All committers have signed the CLA.

whattherestimefor committed:
Fix localized strings

whattherestimefor committed:
Fix build server error

whattherestimefor commented:
After design review, some tweaks:
![Donation screenshots 003](https://github.com/user-attachments/assets/ccac59b6-c638-4385-a7fd-a050eec1e396)

whattherestimefor committed:
Fix minor visual issues

whattherestimefor committed:
Fix build server error better

whattherestimefor committed:
Fixes question of default suggested amount.

Amounts will not necessarily be returned from the server in order. The first amount returned is taken as the default and the amounts are sorted before display.

whattherestimefor committed:
Fixes default campaign should always be expected for the menu item.

whattherestimefor committed:
Fix preferred order of donation frequency.

whattherestimefor committed:
Fix localizations

whattherestimefor committed:
Add memoing of previously dismissed and contributed donation campaigns.

whattherestimefor committed:
Add donations testing helpers to Settings.
